.github/workflows/java-publish.yml

```diff
@@ -0,0 +1,101 @@
+name: Build and publish Java packages
+on:
+  release:
+    types: [released]
+  pull_request:
+    paths:
+      - .github/workflows/java-publish.yml
+
+jobs:
+  macos-arm64:
```

Comment:
i felt like it should not be 3 jobs. but instead we should have 3 run matrix and at the very end we need to copy all 3 share lib (linux-x86, linux-arm, macos-arm) into the same fat jar? 

this will still release 3 different JAR IIUC?

Reply:
yes, if we call `mvn deploy` in three different job, we have to release 3 different jars (with different names?). 
I am not sure that mvn will resolve dependency based on the system, like what pypi does.

Reply:
>we should have 3 run matrix and at the very end we need to copy all 3 share lib (linux-x86, linux-arm, macos-arm) into the same fat jar?

Put them into separate jobs, so that i can pass `upload-artifcat/download-artifcat` between the jobs. If we just run 3 parallel metrics, we need the 4th one to download the artifact?